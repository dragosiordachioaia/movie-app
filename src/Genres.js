import React, { Component } from "react";

import "./Genres.scss";

import GenreItem from "./GenreItem";

export default class Genres extends Component {
  constructor(props) {
    super(props);

    this.displayGenres = this.displayGenres.bind(this);
  }

  displayGenres() {
    return this.props.genres.map(genreData => {
      return (
        <GenreItem
          key={genreData.id}
          {...genreData}
          onCheck={this.props.onGenreCheck}
        />
      );
    });
  }

  render() {
    return (
      <div>
        <p>Genres:</p>
        <ul className="genres-filter">{this.displayGenres()}</ul>
      </div>
    );
  }
}
