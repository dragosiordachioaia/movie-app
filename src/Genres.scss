@import "variables";

.genres-filter {
  list-style-type: none;
  padding-left: 0;

  .genre-item {
    cursor: pointer;
    margin-bottom: 10px;

    &:hover {
      color: $color-secondary;
    }

    label {
      vertical-align: middle;
    }
  }
}
