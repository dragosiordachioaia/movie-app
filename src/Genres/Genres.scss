@import "../variables";

.genres-filter {
  list-style-type: none;
  padding-left: 0;

  .genre-item {
    cursor: pointer;
    position: relative;
    margin-bottom: 10px;
    animation-name: slide-in;
    animation-duration: 0.4s;
    animation-fill-mode: forwards;
    animation-iteration-count: 1;
    opacity: 0;

    @for $i from 1 through 30 {
      &:nth-child(#{$i}) {
        animation-delay: $i * 0.05s + 0.4s;
      }
    }

    @keyframes slide-in {
      from {
        left: -2em;
        opacity: 0;
      }
      to {
        left: 0;
        opacity: 1;
      }
    }

    &:hover {
      color: $color-secondary;
    }

    label {
      font-weight: 700;
      font-size: 0.75em;
      position: relative;
      top: -0.65em;
      left: 0.5em;
    }
  }
}
