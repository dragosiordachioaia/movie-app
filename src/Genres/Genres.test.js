import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Genres from "./Genres";

jest.mock("GenreItem/GenreItem", () => "GenreItem");

let component;
let instance;

beforeEach(() => {
  component = renderer.create(<Genres genres={[]} />);
  instance = component.getInstance();
});

it("renders without crashing", () => {
  instance.props = {
    genres: [],
  };
  const result = instance.render();
  expect(result).toMatchSnapshot();
});

it("renders the right number of genre items", () => {
  instance.props = {
    genres: [{ id: 1 }, { id: 2 }, { id: 3 }],
  };
  const result = instance.render();
  expect(result).toMatchSnapshot();
});
