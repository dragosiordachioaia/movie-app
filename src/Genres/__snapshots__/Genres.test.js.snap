// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the right number of genre items 1`] = `
<div>
  <p>
    Genres:
  </p>
  <ul
    className="genres-filter"
  >
    <GenreItem
      id={1}
    />
    <GenreItem
      id={2}
    />
    <GenreItem
      id={3}
    />
  </ul>
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  <p>
    Genres:
  </p>
  <ul
    className="genres-filter"
  />
</div>
`;
